include/arch/riscv/abstract-branch-instruction-node.hpp

```diff
@@ -274,12 +284,12 @@ class AbstractBranchInstructionNode : public InstructionNode {
   ValidationResult
   _validateResultingProgramCounter(MemoryAccess& memoryAccess) const {
     auto programCounter = riscv::loadRegister<UnsignedWord>(memoryAccess, "pc");
-    auto offset = super::template _getChildValue<UnsignedWord>(memoryAccess, 1);
+    auto offset = super::template _getChildValue<UnsignedWord>(memoryAccess, 2);
     // Check if the program counter would underflow or overflow
-    if (!riscv::isAddressValid(memoryAccess, programCounter + 2 * offset)) {
+    if (!riscv::isAddressValid(memoryAccess, programCounter + 2 * offset, programCounter)) {
```

Comment:
Can you rename `isAddressValid` to `addressIsValid`, I think it makes more sense